docs/pages/frameworks/vue.md

```diff
@@ -35,35 +35,22 @@ If you'd rather not use the CDN for assets, you can create a build task that cop
 
 ## Configuration
 
-You'll need to tell Vue to ignore Shoelace components. This is pretty easy because they all start with `sl-`.
-
-```js
-import { fileURLToPath, URL } from 'url';
-
-import { defineConfig } from 'vite';
-import vue from '@vitejs/plugin-vue';
-
-// https://vitejs.dev/config/
-export default defineConfig({
-  plugins: [
-    vue({
-      template: {
-        compilerOptions: {
-          isCustomElement: tag => tag.startsWith('sl-')
-        }
-      }
-    })
-  ],
-  resolve: {
-    alias: {
-      '@': fileURLToPath(new URL('./src', import.meta.url))
-    }
-  }
-});
-```
+If you haven't configured your Vue.js project to work with custom elements/web components, follow [the instructions here](https://vuejs.org/guide/extras/web-components.html#using-custom-elements-in-vue) based on your project type to ensure your project will not throw an error when it encounters a custom element.
 
 Now you can start using Shoelace components in your app!
 
+## Types
+
+Once you have configured you application for custom elements, you should be able to use Shoelace in your application without it causing any errors. Unfortunately, this doesn't register the custom elements to behave like components built using Vue. To provide autocomplete information and type-safety for your components, you can import the Shoelace Vue types into your `tsconfig.json` to get better integration in your standard Vue and JSX templates.
+
+```json
+{
+  "compilerOptions": {
+    "types": ["@shoelace-style/shoelace/dist/types/vue"]
+  }
+}
+```
```

Comment:
I'd caution heavily against this. It overwrites a lot of expected conventions around types.

It's generally better in the `App.d.ts` to import the types there for global use.

```
// App.d.ts
import "@shoelace-style/shoelace/dist/types/vue"
```

can't remember if it wipes out built in types

<https://www.typescriptlang.org/tsconfig#types>
> If `types` is specified, only packages listed will be included in the global scope. For instance:
> ...
> This tsconfig.json file will only include ./node_modules/@types/node, ./node_modules/@types/jest and ./node_modules/@types/express. Other packages under node_modules/@types/* will not be included.

I've been burnt by this foot gun too many times.

Reply:
Oh, interesting! Do you have some docs you can point me to?

Reply:
For the Vue types, I mean. It looks like you're right about the "compilerOptions" route. That may end up excluding other types.

Reply:
@break-stuff heres the recommendation from the Vue docs:

https://vuejs.org/guide/typescript/options-api.html#augmenting-global-properties

Reply:
It says to add it to the `types` property...

> We can put this type augmentation in a .ts file, or in a project-wide *.d.ts file. Either way, make sure it is included in tsconfig.json. For library / plugin authors, this file should be specified in the types property in package.json.

Reply:
@break-stuff I tried asking Twitter. No luck. So I opened a discussion on Vue. No need holding up the PR for it, we can always fix it in the future.